### ISSUE TITLE
Various TS issues/questions

### ISSUE THREAD

**Issue by easternmotors** [CesiumGS/cesium#8927]
Thanks for 1.70 and the new TS support!

I am working through getting rid of my own Cesium.d.ts and using the new built-in version. Here's a few things I've found when switching over. I tried to look a few few other threads and found some helpful advice/PR's that are already coming. Apologies in advance if than of these are duplicate issues:

**1. EntityCollection.getById should return Entity | undefined**
Definition doesn't match the description/functionality
<img width="425" alt="Screen Shot 2020-06-07 at 1 26 20 PM" src="https://user-images.githubusercontent.com/50411814/83975580-8cf1cc80-a8c2-11ea-88ea-28d6a2a2e179.png">

**2. How do I update a `PositionProperty`?**
I am using `Entity.position` which can be `PositionProperty | undefined`. I saw in another thread that for something like `PolylineGraphics.width` you can use `ConstantProperty(10)` for example. What is the correct workaround for Cartesian3's? Right now I have a few lines that looks something like `entity.position = new Cartesian3(x, y, z)`. Not sure what to do since it doesn't have parameters or a setter.

**3. `Property 'rotationMatrixFromPositionVelocity' does not exist on type 'typeof Transforms'.`**
I looked at the code and saw it was private. Just wondering why?

**4. Can't find a definition anywhere for an `Event` that contains `endPosition`**
In one of my `ScreenSpaceEventHandler` actions, I pass in an `action` that contains `endPosition`. Short repro:
```
(event) => { console.warn(event.endPosition) }
```
I am trying to define what `endPosition` is but can't find an exported event type.

**5. `Entity.position` type doesn't include `CallbackProperty` but works with `CallbackProperty`**
`Type 'CallbackProperty' is not assignable to type 'Cartesian3 | PositionProperty | undefined'.
  Type 'CallbackProperty' is missing the following properties from type 'PositionProperty': referenceFrame, getValueInReferenceFrame ts(2322)`

**6. `PolylineArrowMaterialProperty` parameter should be `color?: Color` not `color?: Property | undefined`?**
From what I've seen, parameters and constructors usually aren't `Property`. It's only when you access them after you create them.

**7. `PolylineGlowMaterialProperty/StripeMaterialProperty` constructor options shouldn't be `Property`?**
Basically same as 6. It feels weird when the constructor is asking for `Property`'s

Thanks again for all the work it took to get this feature in!!!

**Comment by mramato** [comment]
Thanks, @easternmotors! I greatly appreciate you taking the time to write up this issue.

> 1. EntityCollection.getById should return Entity | undefined

Agreed, will fix ASAP.

> 2. How do I update a `PositionProperty`?

For a static position there is a `ConstantPositionProperty`:

```ts
entity.position = new ConstantPositionProperty(new Cartesian3(0, 1, 2));
```

There are several PositionProperty implementations which can be used for various situations, for example if you had time-tagged positions or are receiving real-time tagged data, use `SampledPositionProperty`.  Filtering for `Property` in the [API reference doc](https://cesium.com/docs/cesiumjs-ref-doc/) will get you a full list of all Property types.

> 3. `Property 'rotationMatrixFromPositionVelocity' does not exist on type 'typeof Transforms'.`

This is another case of `Transforms.rotationMatrixFromPositionVelocity` not being part of the CesiumJS API, it's marked `@private` in the code.  I'm not exactly sure why it's marked private and I assume we could probably expose it; but it probably will require adding some tests for it.  Would you be interested in contributing a Pull Request for this?  If not, I can write up an issue so that we can look at exposing it sometime soon.

> 4. Can't find a definition anywhere for an `Event` that contains `endPosition`

This looks like an issue with missing/incomplete documentation on our side.  We don't actually concretely define what the callback functions called `ScreenSpaceEventHandler` are.  There are a bunch of xxxEvent objects defined in `ScreenSpaceEventHandler.js`, most simple have a `position` property, but some have other information.  I'm going to write up a separate issue for this one since it is a non-trivial amount of work to fix but definitely something we should do.

> 5. Entity.position type doesn't include CallbackProperty but works with CallbackProperty

It only works with callback property "by accident" in some parts of the code.  The position interface requires a `referenceFrame` be defined and the `getValueInReferenceFrame` method be implemented.  That being said, you are not the first person to ask about this and we want to fix it, most likely by adding a CallbackPositionProperty for CesiumJS 1.71.0

> 6. `PolylineArrowMaterialProperty` parameter should be `color?: Color` not `color?: Property | undefined`?

This one has already been fixed in master, it's `{Property|Color}`, since we don't need to specifically define `undefined` for parameters and both `Property` and `Color` work.

> 7. `PolylineGlowMaterialProperty/StripeMaterialProperty` constructor options shouldn't be `Property`?

Similar to 6, and also already fixed in master

So just to recap;

* A fix for 1 will be part of CesiumJS 1.70.1 later this week.
* 2 has an explicit answer
* 3, 4, and 5 require some additional work but are all stuff we could/should do.
* 6 and 7 are already fixed in master. :tada: 

Thanks again for the write-up @easternmotors!  Hopefully I answered all of your questions so far.

I'll probably write up separate issues for 3, 4 and 5 once and comment here so that they get the attention they deserve but let us know if you are interested in working on any of them with us.

**Comment by easternmotors** [comment]
@mramato Awesome, thanks for the quick reply and the helpful explanations! I can take a look at what it would entail to expose `Transforms.rotationMatrixFromPositionVelocity` definitely.

**Comment by easternmotors** [comment]
@mramato Let me know what you think of the PR for bullet 3. It's my first time contributing but I think I followed all of the helpful docs you have written up for contributing.
